onnxruntime/core/framework/allocatormgr.cc

```diff
@@ -54,4 +61,38 @@ AllocatorPtr CreateAllocator(const AllocatorCreationInfo& info) {
   return AllocatorPtr(std::move(device_allocator));
 }
 
+
+// Update allocator in the provider if already present; ignore if not.
+void AllocatorManager::ReplaceAllocator(AllocatorPtr allocator) {
+  const auto& info = allocator->Info();
+  auto ite = mem_info_set_.find(info);
+  if (ite != mem_info_set_.end()) {
+    const int key = MakeKey(info.id, info.mem_type);
+    allocators_[key] = allocator;
+  }
+}
+
+void AllocatorManager::InsertAllocator(AllocatorPtr allocator) {
+  const OrtMemoryInfo& info = allocator->Info();
+  auto ite = mem_info_set_.find(info);
+  if (ite != mem_info_set_.end()) {
+    LOGS_DEFAULT(WARNING) << "duplicated allocator: " << info.ToString();
+    return;
+  }
+  const int key = MakeKey(info.id, info.mem_type);
+  allocators_.insert({key, allocator});
+  mem_info_set_.insert(ite, info);
+  allocator_list_.push_back(allocator);
+}
+
+AllocatorPtr AllocatorManager::GetAllocator(int id, OrtMemType mem_type) const {
+  if (allocators_.empty()) {
+    return nullptr;
+  }
```

Comment:
Is this needed? I thought find() would handle empty container.

Reply:
removed

---
In reply to: [564296547](https://github.com/microsoft/onnxruntime/pull/6332#discussion_r564296547) [](ancestors = 564296547)